### DISCUSSION TITLE
How to access "root custom element"?

### DISCUSSION THREAD

**Discussion by dbwodlf3** [lit/lit#2653] [Q&A] [answered]
I would like to change style of custom element tag from event in shadow elements.

```html
<my-element></my-element> => <my-element style="display:none"></my-element>
```

But, i don't know how to access root element.

renderRoot, getRootNode and shadowRoot are not html element instance...

how can access root of custom component?

**Comment by bicknellr** [comment, marked as answer]
> renderRoot, getRootNode and shadowRoot are not html element instance...

Maybe you're looking for `.host`? Shadow roots have a `.host` property that gives you the element that hosts it: https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot/host

However, if you're actually trying to modify (e.g.) a `<my-element>`'s styles from within the class defining `<my-element>` itself, then you can just modify `this` directly because `this` literally is the `<my-element>`.

**Reply by dbwodlf3** [reply]
oh. thanks. "just this".
